### PR TITLE
Chattr

### PR DESCRIPTION
**Pull request by benschac**
Starts Node Chat project.

### CODE REVIEW

File: front-end-dev-book/chattr/app/scripts/src/app.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class ChatApp {
+  constructor({
+    message: m,
+    user: u="batman",
+    timestamp: t=(new Date().getTime())
+    // Deconstruct and use default parameters.
+  }) {
+    this.message = m;
+    this.user = u;
+    timestamp: t
```
Comment (stanzheng) [inline front-end-dev-book/chattr/app/scripts/src/app.js]: I didn't run this code but should this have a this on timestamp here or is that intentional?

File: front-end-dev-book/chattr/app/scripts/src/app.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+class ChatApp {
+  constructor({
+    message: m,
+    user: u="batman",
+    timestamp: t=(new Date().getTime())
+    // Deconstruct and use default parameters.
```
Comment (stanzheng) [inline front-end-dev-book/chattr/app/scripts/src/app.js]: Where can I find out more about this constructor pattern? I'm rusty but I'd like to find out more what this is doing.

Comment (benschac) [inline front-end-dev-book/chattr/app/scripts/src/app.js]: It's a new feature in ES6, so I don't think you're rusty :D heh. It's a combination of array deconstruction and default params.  Check out: https://es6cheatsheet.com/#destructuring and  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

**Comment by benschac** [comment]
Hmmmmm, I'm not a 100%.  I'm going to play with it now.

File: front-end-dev-book/chattr/app/scripts/src/app.js (hunk 2 above)
Comment (stanzheng) [inline front-end-dev-book/chattr/app/scripts/src/app.js]: ah i knew about the Class syntax and destructuring but not seen them mixed together before along with the variable assignment it threw me for a loop too.

Thanks for explaining!

**Pull request merged**